src/LibraryManager.Contracts/FileHelpers.cs

```diff
@@ -0,0 +1,219 @@
+            {
+                DirectoryInfo dir = Directory.CreateDirectory(directoryPath);
+                using (FileStream destination = File.Open(fileName, FileMode.Create, FileAccess.Write, FileShare.None))
+                {
+                    await sourceStream.CopyToAsync(destination);
+                }
+
+                FileInfo fileInfo = new FileInfo(fileName);
+                Debug.Assert(fileInfo.Length != 0, string.Format("File {0} is empty", fileName));
+
+                return true;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Reads from a file a returns its content as text
+        /// </summary>
+        /// <param name="fileName"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public static async Task<string> ReadFileAsTextAsync(string fileName, CancellationToken cancellationToken)
+        {
+            using (Stream s = await ReadFileAsStreamAsync(fileName, cancellationToken).ConfigureAwait(false))
+            using (var r = new StreamReader(s, Encoding.UTF8, true, 8192, true))
+            {
+                return await r.ReadToEndAsync().WithCancellation(cancellationToken).ConfigureAwait(false);
+            }
+        }
+
+        /// <summary>
+        /// Reads from a file a returns its content as Stream
+        /// </summary>
+        /// <param name="fileName"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public static Task<Stream> ReadFileAsStreamAsync(string fileName, CancellationToken cancellationToken)
+        {
+            return Task.FromResult<Stream>(new FileStream(fileName, FileMode.Open, FileAccess.Read, FileShare.Read, 1, useAsync: true));
+        }
+
+        /// <summary>
+        /// Copies a file from source to destination
+        /// </summary>
+        /// <param name="sourceFile"></param>
+        /// <param name="destinationFile"></param>
+        /// <returns></returns>
+        public static bool CopyFile(string sourceFile, string destinationFile)
+        {
+            try
+            {
+                File.Copy(sourceFile, destinationFile, true);
+
+                return true;
+            }
+            catch (Exception)
+            {
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Returns true if both files have same same, size and last written time
+        /// </summary>
+        /// <param name="firstFile"></param>
+        /// <param name="secondFile"></param>
+        /// <returns></returns>
+        public static bool AreFilesUpToDate(FileInfo firstFile, FileInfo secondFile)
+        {
+            if (firstFile == null || secondFile == null || !firstFile.Exists || !secondFile.Exists)
+            {
+                return false;
+            }
+
+            if (string.Compare(firstFile.Name, secondFile.Name, true) != 0 ||
```

Comment:
> Also, do we need to be case-sensitive on non-Windows platforms?

Not always. 
Mac OS by default is case insensitive.  
dotnet core fx has an open feature request to provide an API to determine this 
https://github.com/dotnet/corefx/issues/1086

This has been open since 2015 though.